src/main.c

```diff
@@ -147,99 +136,36 @@ int main(void) {
 #ifdef DEV_BUILD
   ekp_queue_init();
 #endif
-
   application_init();
 
 #ifdef RUN_ENGINE
-  logo_scr_init(2000);
-
-  while (1) {
-    engine_ctx_t *main_engine_ctx = get_core_flow_ctx();
-    engine_run(main_engine_ctx);
-  }
-#else /* RUN_ENGINE */
-
 #if USE_SIMULATOR == 0
   if (fault_in_prev_boot()) {
     handle_fault_in_prev_boot();
   } else
-#endif    // USE_SIMULATOR
+#endif /* USE_SIMULATOR == 0 */
   {
     logo_scr_init(2000);
     device_provision_check();
-    reset_flow_level();
-#if X1WALLET_MAIN
-    if (device_auth_check() == DEVICE_AUTHENTICATED)
+
+    if (MANAGER_ONBOARDING_STEP_COMPLETE == get_onboarding_step() &&
+        DEVICE_AUTHENTICATED == get_auth_state()) {
       check_invalid_wallets();
-#endif
+    }
+  }
+
+  while (1) {
+    engine_ctx_t *main_engine_ctx = get_core_flow_ctx();
+    engine_run(main_engine_ctx);
   }
+#else /* RUN_ENGINE */
 
   while (true) {
     if (keypad_get_key() != 0)
       reset_inactivity_timer();
-    // Flow
-    main_app_ready = true;
-    if (CY_Read_Reset_Flow()) {
-      if (!CY_reset_not_allowed()) {
-        cy_exit_flow();
-      } else {
-        sys_flow_cntrl_u.bits.reset_flow = false;
-      }
-    }
-
-    if (sys_flow_cntrl_u.bits.nfc_off == false) {
-      nfc_deselect_card();
-    }
-
-    if (counter.next_event_flag != 0) {
-      PRINT_FLOW_LVL();
-      mark_device_state(
-          CY_UNUSED_STATE,
-          counter.level < LEVEL_THREE ? 0 : flow_level.level_three);
-      reset_next_event_flag();
-#if X1WALLET_MAIN
-      level_one_tasks();
-#elif X1WALLET_INITIAL
-      level_one_tasks_initial();
-#else
-#error Specify what to build (X1WALLET_INITIAL or X1WALLET_MAIN)
-#endif
-    }
-
-#if USE_SIMULATOR == 1
-    usbsim_continue_loop();
-#endif    // USE_SIMULATOR
```

Comment:
Since we don't use the usb rn with simulator, we can remove this, but let's restore this just for reference for when we have to move it to the get events task.

Reply:
Restored